website/content/en/guides/developer/debugging/debugging.md

```diff
@@ -0,0 +1,760 @@
+---
+title: A guide for debugging Vector pipelines
+short: Debugging Vector
+description: Step by step guide for debugging pipelines
+tags: [ "dev", "debugging", "guides", "guide" ]
+author_github: https://github.com/pront
+domain: dev
+aliases: [ "/docs/guides/developer/config-autocompletion.md" ]
+weight: 1
+---
+
+# Debugging Guide
+
+This guide will describe an imaginary use case and how to progressively resolve issues using dev tools and Vector commands.
+
+Imagine we have two sources, and we want to assign group IDs to Vector logs based on the source. Finally, we want to send those modified
+logs to an HTTP server for further processing and/or storage.
+
+In the following sections we will examine the tools we have at our disposal.
+
+<img src="/img/guides/debugging-meme.png"  alt="debugging-meme" width="400"/>
+
+## Vector Tools
+
+### Vector top
+
+Use [vector top](https://vector.dev/docs/reference/cli/#top) to display topology and metrics in the console.
+For example, you can inspect how many events are produced by sources vs how many events reach the sinks.
+
+Command:
+
+```shell
+vector top
+```
+
+Screen:
+
+<img src="/img/guides/vector-top.png"  alt="top-screen"/>
+
+### The internal metrics source
+
+The [internal_metrics](https://vector.dev/docs/reference/configuration/sources/internal_metrics/) source can be used to inspect component
+metrics in detail.
+
+### Vector Tap
+
+With [vector tap](https://vector.dev/guides/level-up/vector-tap-guide/) you see the input and/or outputs of your components.
+Here, it is worth mentioning a caveat, sinks don't have outputs thus we cannot tap them. This means we cannot inspect actual payloads that
+are sent downstream.
+
+Command:
+
+```shell
+vector tap
+```
+
+Screen:
+
+<img src="/img/guides/vector-tap.png"  alt="tap-screen"/>
+
+### Console
+
+Note that the [console sink](https://vector.dev/docs/reference/configuration/sinks/console/) can also be very useful here. For example:
+
+```yaml
+my_console_sink:
+  type: console
+  inputs:
+    - my_source_0
+    - my_source_1
+  encoding:
+    codec: json
+    json:
+      pretty: true
+```
+
+## Python HTTP Server
+
+The following code implements a Python server that imitates a downstream system to which Vector publishes logs. We will use this server
+to demonstrate a few scenarios.
```

Comment:
This seems like something that might be nice to have under test in the future. That is, have the script be a Python file in the repo and have a test that actually exercises it to ensure it doesn't break.